packages/core/lib/segments/attributes/remote_request_data.js

```diff
@@ -14,7 +14,7 @@ function RemoteRequestData(req, res, downstreamXRayEnabled) {
 
 RemoteRequestData.prototype.init = function init(req, res, downstreamXRayEnabled) {
   this.request = {
-    url: (req.agent.protocol + '//' + req.getHeader('host') +  stripQueryStringFromPath(req.path)) || '',
+    url: ((req.agent && req.agent.protocol) + '//' + req.getHeader('host') +  stripQueryStringFromPath(req.path)) || '',
```

Comment:
While this avoids throwing the error, in the event that `req.agent` is undefined it would print the URL as:

```
undefined//example.com/
```

I'd advise changing this to: 

```js
url: (req.agent && req.agent.protocol) ? (req.agent.protocol + '//' + req.getHeader('host') +  stripQueryStringFromPath(req.path)) : '',	
```

Reply:
Done; looking at the original line again, I see that it contained an undefined check that would never be executed.